apps/publish/content/common.py

```diff
@@ -994,15 +1003,28 @@ def _update_picture_metadata(self, updates, original, updated):
                 if not media_id:
                     continue
 
-                original_metadata = get_metadata_from_item(original, mapping)
-                metadata = get_metadata_from_item(updated, mapping)
+                picture = app.media.get(media_id)
+                binary = picture.read()
 
-                if original_metadata == metadata:
+                file_metadata = (
+                    video_read_metadata(binary) if updated[ITEM_TYPE] == "video" else image_read_metadata(binary)
```

Comment:
I think it would simplify the usage in general if we would have a generic functions with type param, so this logic could be inside:
```py
file_metadata = read_metadata(binary, updated[item_type])
```
for all those related methods